### PR TITLE
[ECO-1809] add pools page

### PR DESCRIPTION
**Pull request by CRBl69**
# Description

This PR adds the connections to Inbox and the Aptos blockchain for the pools page.

# Testing

Deploy the website and test.

Note that the inbox submodule is not up to date, so it will not work until it is.

# Checklist

- [x] Did you check all checkboxes from the linked Linear task?

### CODE REVIEW

File: src/typescript/frontend/src/lib/hooks/queries/use-simulate-provide-liquidity.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      quoteAmount.toString(),
+    ],
+    queryFn: () =>
+      invalid
+        ? {
+            base_amount: "0",
+          }
+        : simulateProvideLiquidity({ aptos, ...args, marketAddress, quoteAmount }),
+    staleTime: Infinity,
+  });
```
Comment (xbtmatt) [inline src/typescript/frontend/src/lib/hooks/queries/use-simulate-provide-liquidity.ts:48]: It looks like you're using both `swr` and `tanstack`, although I'm probably peeking in on this a bit early so ignore me if you were planning on removing `swr` already

Just flagging that we can just stick to tanstack since they serve the same purpose 😄

Comment (CRBl69) [reply src/typescript/frontend/src/lib/hooks/queries/use-simulate-provide-liquidity.ts:48]: My bad, I didn't know about that tool.

File: src/typescript/frontend/src/components/pages/pools/ClientPoolsPage.tsx [hunk 2]
```diff
@@ -16,9 +18,23 @@ import {
...
+    .then((res) => res.text())
+    .then((txt) => parseJSON(txt));
 
 export const ClientPoolsPage = () => {
+  const [sortBy, setSortBy] = useState<SortByPageQueryParams>("all_time_vol");
+  const [orderBy, setOrderBy] = useState<"desc" | "asc">("desc");
+  const [selectedIndex, setSelectedIndex] = useState<number>();
+  const { data, error, isLoading } = useSWR(
+    `/pools/api?sortby=${sortBy}&orderby=${orderBy}`,
+    fetcher
```
[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/typescript/frontend/src/components/pages/pools/ClientPoolsPage.tsx:35]: Yes, exactly! Basically what you already have with `swr` can go into a server action and be passed down from server component to the client as prop data, and then you can use `tanstack` to manage any custom querise for the user.

You could also just put the `unstable_cache` (in the form of `cached` in our repo, sorry for any alias confusion...) call in a server action and then use that in the client, too.

The primary motivator for this is lots of users are going to sort and get the same exact data when querying with the pools filters- so we can cache it for some X amount of seconds per user instead of having each of them call the postgrest API independently.

Basically, at least from my understanding, with `unstable_cache` the nextjs server acts as middleware and stores the request in the Data Cache rather than it being stored in a client-side cache with tanstack/swr

https://nextjs.org/docs/app/building-your-application/caching

[the remark this answers is not in the document]
Comment (xbtmatt) [reply src/typescript/frontend/src/components/pages/pools/ClientPoolsPage.tsx:35]: Theoretically you should be able to use the server action in a client component actually- but I've run into a lot of issues trying to get it to work. I think it's because we're using React 18 instead of the canary releases (react 19 and nextjs 15)

**Review by xbtmatt — APPROVED** [review]
